packages/engine/Source/Core/Resource.js

```diff
@@ -568,32 +533,55 @@ Resource.prototype.getUrlComponent = function (query, proxy) {
     return this._url;
   }
 
-  const uri = new Uri(this._url);
+  let url = query
```

Comment:
This may be bikeshedding a bit, but this ternary may be a bit obtuse.
```
let url = this._url;
if (query) {
    url = `${url}${stringifyQuery(this.queryParameters)}`
}
```
seems more readable (and therefore more maintainable) to me.